hexapod_controller/src/hexapod_controller/hexapod_controller/body.py

```diff
@@ -80,225 +106,78 @@ def send_data(self):
-                
-                self.body_IK_.publish(cmd)
-                index += 1
-                
-        elif (self.g_group_1_legs_up): 
-            # self.group_1_walk(msg)
-            for y in points:
-                time.sleep(sleep_time)
-                '''
-                forward: 
-                    leg_1: x > 0
-                    leg_3: x < 0
-                    leg_5: x > 0
-                
-                backward: 
-                    leg_1: x < 0
-                    leg_3: x > 0
-                    leg_5: x < 0
-                '''
-                
-                if index_2 > len(points) / 2 - 1 :
-                    variant_2 = 1
-                    self.g_group_2_legs_up = 1
-                else:
-                    self.g_group_2_legs_up = 0
-                
-                leg_value_2 = ik.leg_ik_test(y * x_sign, 1, variant_2)
-                leg_value_4 = ik.leg_ik_test(-y * x_sign, 5, variant_2)
-                leg_value_6 = ik.leg_ik_test(y * x_sign, 3, variant_2)
+    def group_walk(self):
+        '''
+         Function dedicated to scheduling leg sequences depending on gait
+        '''
+        # constants
+        GITE_SIZE = 6
+        NUMBER_OF_LEGS = 6
+        
+        # variables
+        x = 1 # <-- move forward, change to -1 to go backward
+        
+        # iterating over gait table - column
+        for gait_index in range(GITE_SIZE):
+            # iterating over legs gait table - rows
+            for gait_leg in range(NUMBER_OF_LEGS):
+                self.leg_trajectory(id=gait_leg, forward=x, up_or_down=self.leg_gait_status[gait_leg][gait_index]) # FLAGA BLEDU
+            
+            
+    def leg_trajectory(self, id, forward, up_or_down):
+        '''
+         Function dedicated to move specyfic leg in meaning of walking forward / backward
+         id = id of leg
+         x = forward -> 1, backward -> -1
+         up_or_down = if 1 it means that leg should be in air
+         rot_angle = trajectory line rotation in space, for future development, hexa rotation
+        '''
+        
+        # empty msg type initialization, with default values
+        cmd = Leg()
+        
+        # variables
+        variant     = 0
+        WAIT_TIME   = 0.3                                   # FLAGA BLEDU zwiekszyc?
+        
+        # step logic
+        if up_or_down:
+            variant     = 0
+            points      = [point for p in range(20,-25,-5)]  # FLAGA BLEDU
+        else:
+            variant     = 1
+            points      = [point for p in range(-20,25,5)]   # FLAGA BLEDU tak listować punkty?
+            
+        # leg inverse kinematics calculations and msg publications
+        # do this for every point in trajectory
+        for x in points:
+            # give some time for hardware to move
+            time.sleep(WAIT_TIME)
+
+            leg_value = ik.leg_ik(x*forward, id, variant)      # FLAGA BLEDU x * forward?
```

Comment:
Exactly. We did use "x * x_sign" previously but now our "x_sign" became "forward".